x/emissions/migrations/v4/migrate.go

```diff
@@ -6,7 +6,8 @@ import (
 	storetypes "cosmossdk.io/store/types"
 	alloraMath "github.com/allora-network/allora-chain/math"
 	"github.com/allora-network/allora-chain/x/emissions/keeper"
-	oldtypes "github.com/allora-network/allora-chain/x/emissions/migrations/v3/types"
+	v2types "github.com/allora-network/allora-chain/x/emissions/migrations/v3/types"
+	v3types "github.com/allora-network/allora-chain/x/emissions/migrations/v4/types"
```

Comment:
you call it v3types but pointing at v4 ?

Reply:
yes correct, that is intentional

The types for v3, which is v0.4.0, which is now _in the past_ are stored in the migration folder for the v4 migration, because the v4 migration needs to remember the types from the past in order to migrate.
The types for v2, which is v0.3.0, which is _two migrations ago_ are stored in the migration folder for the v3 migration, because the v3 migration needs to remember the types from the past. In addition, because we messed up doing the v3 migration, now the v4 migration ALSO needs to reference those old types. Hence the naming.

I will rename it further to make it even more explicit.